### PR TITLE
Use openssl instead of mcrypt.

### LINKED ISSUE #32: Mcrypt is abandoned, use OpenSSL
Making a note for my own fork.

### PR DESCRIPTION
**Pull request by sarciszewski**
Resolves #32

### CODE REVIEW

**Comment by narfbg** [comment]
@sarciszewski No immediate problems at first glance, but you should just set `CIPHER` to 'aes-128' instead of computing `$method` dynamically.

**Comment by sarciszewski** [comment]
Yeah, I'll send an updated commit tonight :)

**Comment by sarciszewski** [comment]
Test fails on PHP 5.3 which was EOL'd months ago due to an undefined constant (`OPENSSL_RAW_DATA`). http://php.net/eol.php
Test fails on PHP 5.2.

I recommend, in the spirit of #36 support be enforced for a minimum of PHP 5.4.

For users who are on 5.2/5.3, I recommend branching/tagging before merging so there's an mcrypt version lying around.

File: Crypto.php [hunk 1]
```diff
@@ -304,42 +305,22 @@ private static function PlainEncrypt($plaintext, $key, $iv)
...
+            $method,
+            $key,
+            OPENSSL_RAW_DATA,
+            $iv
+        );
         if ($plaintext === FALSE) {
             throw new CannotPerformOperationException();
         }
-
+
```
Comment (defuse) [inline Crypto.php]: Evil whitespace!

**Comment by tsyr2ko** [comment]
*SecureRandom* function still uses mcrypt_* function.
Should use *openssl_random_pseudo_bytes* instead

**Comment by sarciszewski** [comment]
Is `openssl_random_pseudo_bytes()` safe to use?

**Comment by narfbg** [comment]
As long as you use its second parameter to check if a strong-enough source was used, I regard it as better than fopen()-ing /dev/urandom and trying to do it all yourself.

**Comment by sarciszewski** [comment]
I meant from a cryptographer's perspective. I managed to get to `RAND_SSLeay()` in the openssl codebase before getting a headache.

**Pull request merged**